docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+      output that follows each function call.
+      """
+      if np.isinf(a) or np.isinf(b):
+          raise ValueError("Cannot perform substraction operations involving infinity.")
+
+      warnings.warn("The subtract function encountered a nan value.", UserWarning)
+
+      return b - a if switch_order else a - b
+
+Template docstring
+~~~~~~~~~~~~~~~~~~
+
+This template docstring may be copied into new functions. Usually only
+some of the sections will be necessary for a particular function, but
+any sections that are included should be in the order provided.
+
+.. code-block:: python
+  :caption: Docstring template
+
+  def sample_function():
+      r"""
+      Return ...
+
+      Parameters
+      ----------
+
+      Returns
+      -------
+
+      Raises
+      ------
+
+      Warns
+      -----
+
+      See Also
+      --------
+
+      Notes
+      -----
+
+      References
+      ----------
+
+      Examples
+      --------
+
+      """
+
+
+Documentation guidelines
+========================
+
+This section contains guidelines and best practices for writing
+PlasmaPy documentation.
+
+* Write documentation to be understandable to students taking their
+  first course or beginning their first research project in plasma
+  science. Include highly technical information only when necessary.
+
+* Use the `active voice <https://en.wikipedia.org/wiki/Active_voice>`_
+  in the present tense.
+
+* Keep the documentation style consistent within a file or module, and
+  preferably across all of PlasmaPy's documentation.
+
+* Refer to the `numpydoc`_ standard for how to write docstrings for
+  classes, class attributes, and constants.
+
+* Update code and corresponding documentation at the same time.
+
+* Write sentences that are simple, concise, and direct rather than
+  complicated, vague, or ambiguous. Prefer sentences with ≲ 20
+  words.
+
+* Avoid idioms, metaphors, and references that are specific to a
+  particular culture.
+
+* Use technical jargon sparingly. Define technical jargon when
+  necessary.
+
+* Many words and software packages have more than one common spelling
+  or acronym. Use the spelling that is used in the file you are
+  modifying, which is preferably the spelling used throughout
+  `PlasmaPy's online documentation`_.
+
+  * More generally, it is preferable to use the spelling that is used
+    in `Python's documentation`_ or the spelling that is used most
+    commonly.
+
+  * Represent names and acronyms for a software package as they are
+    represented in the documentation for that package.
+
+* Write the full namespace when referring to code objects within
+  PlasmaPy. For example, write
+  ``~plasmapy.formulary.parameters.Alfven_speed`` rather than
+  ``~plasmapy.formulary.Alfven_speed``.  This does not need to be done
+  for intersphinx links to other packages; it is fine to do
+  ``~astropy.units.Quantity`` and not necessary to do
+  ``~astropy.units.quantity.Quantity``.
```

Comment:
```suggestion
  ``~plasmapy.formulary.Alfven_speed``.  This does not necessarily
  need to be done when referencing external packages, since
  each package may have their own standard.  For example, Astropy's
  `~astropy.units.Quantity` class is defined in
  ``astropy.units.quantity.Quantity`` but is also indexed at
  ``~astropy.units.Quantity`` so either option will link to some
  documentation.
```